### ISSUE TITLE
Support for java 17/sbt 1.6+

### ISSUE THREAD

**Issue by deevvicom** [davidB/scala-maven-plugin#580]
Upgrading a package to use feature from java 17 fails the compilation because scala maven plugin uses sbt 1.5.

**Comment by slandelle** [comment]
@davidB I can't manage to release. I keep on getting this error:

```
[INFO] [ERROR] Failed to execute goal com.github.github:site-maven-plugin:0.12:site (default) on project scala-maven-plugin: Server 'github' not found in settings -> [Help 1]
```

which is bullshit as I do have one in my settings (and my personal access token is valid with the documented permissions):

```
  <servers>
    <server>
      <id>sonatype-nexus-snapshots</id>
      <username>slandelle</username>
      <password>*****</password>
    </server>
    <server>
      <id>sonatype-nexus-staging</id>
      <username>slandelle</username>
      <password>*****</password>
    </server>
        <!-- to upload site on github
     password is a personal access token https://github.com/settings/tokens
     with permission read:packages, read:user, repo, user:email, write:packages
    -->
    <server>
      <id>github</id>
      <username>slandelle</username>
      <password>*****</password>
    </server>
  </servers>
```

Any clue?

**Comment by slandelle** [comment]
@deevvicom I've released 4.6.0 that ships with zinc 1.6. It's available on maven central.

**Comment by slandelle** [comment]
`-source` and `-target` no longer exist since Java 11. You must use `--release`.
